### ISSUE TITLE
3D Pose with limelight (not an issue, just something I thought you would want to know if you didn't already)

### ISSUE THREAD

**Issue by AlabroX201** [team6637/FRC2023#1]
I would do this with a pull request, but I can't figure out how to use them. This is Alex from Mid Pen (it's my first time) 

If you configure your limelight to use 3d AprilTags, you can get them to output your current Pose on the field. 

To get your 3d pose, use this:

  private double[] botpose = table.getEntry("botpose").getDoubleArray(new double[6]);

This will need to be constantly set/updated in periodic(). botpose[0] is the x pose, botpose[1] is the y pose, and botpose[5] is rotation.

We used this to update our odometry when we see it. I can share that code with you if you want.

P.S. These pose values assume the origin is the center of the field, not the bottom left corner (blue alliance community).  I can get you the offsets if you want

**Comment by ericjanofski** [comment]
Hey, thanks for the input. We aren't using the 3d aspects of April Tags, just the tx offset to center the bot. It's certainly something we would like to look into if there's time. How are you using the pose practically?

**Comment by AlabroX201** [comment]
We are using it for trajectories this year. We had to go with a differential drivetrain. We plan on starting the robot so the limelight facces the AprilTag to get out initial pose on the field. Our Auton goes from there.

**Comment by ericjanofski** [comment]
Oh wow, that's really sweet. Have you worked with Path Planner? that's how we're doing our trajectory generation

**Comment by AlabroX201** [comment]
Yes, We plan on using Path Planner. I just learned how to use the trajectories. I am currently building them in the code, but we plan on using the trajectories from Path Planner. When we tested it, I forgot to pass it a max velocity constraint. Lets just say we popped a wheelie and sideswiped a table before we could disable.

**Comment by ericjanofski** [comment]
Ha, very funny. It's very scary and exciting hitting that enable button. Let me know if you all have any questions. we have auton code up but it's still being developed this week and next.

**Comment by AlabroX201** [comment]
Ok, will do. Thanks!
